YallpAndroid/app/src/main/java/com/example/yallp_android/activities/ProfileActivity.java

```diff
@@ -99,20 +102,54 @@ public void onResponse(Call<UserInfo> call, Response<UserInfo> response) {
                                 //.putString("bio", userInfo.getBio())
                                 .commit();
                     }
-                    else{
-
-                    }
                 }
 
                 @Override
                 public void onFailure(Call<UserInfo> call, Throwable t) {
-                    Log.i("Melih_debug", "profile call didnt work: " + t.getMessage());
                 }
             });
 
-
             editor.putBoolean("newSession", false);
             editor.apply();
         }
     }
+
+    private void logout(){
+        clearApplicationData();
+        SharedPreferences sharedPref = getSharedPreferences("yallp", Context.MODE_PRIVATE);
+        sharedPref.edit().clear().apply();
```

Comment:
both sharedPref and its editor are defined as final in the main onCreate method. To prevent duplicate variable creation, one could add them to logout() method as parameters:

logout(final SharedPreferences sharedPref, final SharedPreferences.Editor editor)

and the part where it is called in onCreate can be updated as

logout(sharedPref, editor);